zebrad/src/components/mempool/storage.rs

```diff
@@ -294,6 +301,56 @@ impl Storage {
             .remove_all_that(|tx| mined_ids.contains(&tx.transaction.id.mined_id()))
     }
 
+    /// Removes and rejects transactions from the mempool that contain any outpoints or nullifiers in
+    /// the `spent_outpoints` or `nullifiers` collections that are passed in.
+    ///
+    /// Returns the number of transactions that were removed and rejected.
+    pub fn reject_invalidated_transactions(
+        &mut self,
+        spent_outpoints: HashSet<OutPoint>,
+        sprout_nullifiers: HashSet<&sprout::Nullifier>,
+        sapling_nullifiers: HashSet<&sapling::Nullifier>,
+        orchard_nullifiers: HashSet<&orchard::Nullifier>,
+    ) -> usize {
+        let mined_ids: HashSet<_> = self
```

Comment:
Renamed to `duplicate_spend_ids` in [`b5cffbf`](https://github.com/ZcashFoundation/zebra/pull/5434/commits/b5cffbfe887e5701fb6b72020c5f390e05319c90) (just moved below the removal of `mined_ids` to avoid needless iteration over spent outpoints and nullifiers of mined transactions.)